File: 0049-group-anagrams/0049-group-anagrams.go
import (
    "sort"
)

type anagram struct {
    key string /* must be sorted */
    members []string
}

func srt(s string) string {
    t := strings.Split(s, "")
    sort.Strings(t)
    return strings.Join(t, "")
} 

func groupAnagrams(strs []string) [][]string {
    
    var a [][]string
    
    d := make(map[string][]int)
    
    for i, e := range strs {
        
        t := srt(e)
        
        d[t] = append(d[t], i)
        
    }
    
    for _, v := range d {
        
        var t []string
        
        for _, e := range v {
            t = append(t, strs[e])
        }
        
        a = append(a, t)
    }
    
    return a
    
    
    
    
    
    
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        d = {}
        
        for i in range(len(strs)):
            st = str(sorted(strs[i]))
                    
            if st not in d:
               d[st] = []
            
            d[st].append(strs[i])
            
        a = []
        
        for e in d:
            a.append(d[e])
            
        return a
        